Configuration Cache Documentation Update for Gradle 9.0 (#33741)
gradle/gradle: platforms/documentation/docs/src/docs/userguide/optimizing-builds/configuration-cache/configuration_cache.adoc

|===

The Configuration Cache is similar to the <<build_cache#build_cache,Build Cache>>, but they store different types of data:

- *Build Cache*: Stores outputs and intermediate files of the build (e.g., task outputs, artifact transform outputs).
- *Configuration Cache*: Stores the build configuration for a particular set of tasks, capturing the output of the configuration phase.

image::configuration-cache-2.png[]

[IMPORTANT]
====
This feature is _not enabled by default_ and has the following limitations:

- It does not yet support all <<configuration_cache_status.adoc#config_cache:plugins:core, Core Gradle Plugins>> and <<configuration_cache_status.adoc#config_cache:not_yet_implemented, features>>. Full support is in progress.
- Your build and its plugins may require modifications to meet the <<configuration_cache_requirements.adoc#config_cache:requirements, Configuration Cache requirements>>.
- <<config_cache:ide,IDE imports and syncs>> do not yet use the Configuration Cache.
- Continuous Integration (CI) support is actively evolving.

> The Configuration Cache will be the preferred mode of execution in Gradle 9.0.

Reviewer: 💅 Can we change this to present tense? And maybe link to something that describes what the "preferred mode" is (I suspect, the blog post is our best source so far).
```suggestion
Since Gradle 9.0 the Configuration Cache is the [preferred mode of execution](https://blog.gradle.org/road-to-configuration-cache#preferred-mode-of-execution).
```
